src/builder.rs

```diff
@@ -326,14 +328,17 @@ impl DoubleArrayAhoCorasickBuilder {
     }
 
     #[inline(always)]
-    fn find_base(&self, labels: &[u8], helper: &BuildHelper) -> u32 {
+    fn find_base(&self, labels: &[u8], helper: &BuildHelper) -> NonZeroU32 {
         for idx in helper.vacant_iter() {
             let base = idx ^ u32::from(labels[0]);
             if self.check_valid_base(base, labels, helper) {
-                return base;
+                if let Some(base) = NonZeroU32::new(base) {
+                    return base;
+                }
```

Comment:
I think it is better to check whether `base != 0` in `check_valid_base` because of its function name.
I suggest to modify `check_valid_base` to return `Option<NonZeroU32>`.

Reply:
fixed